### PR TITLE
Add Markdown custom field type

### PR DESCRIPTION
**Pull request by MeganerdDev**
# Closes: #4006 
# What's Changed

- Added new custom field type for Markdown
  - Reused CommentField
  - Reused SmallTextarea widget -> not required for PR
- Apply markdown render to template

# NOTE
- I think still we need to perform `render_markdown()` to test for script injection similar to [custom_fields.py#L588](https://github.com/nautobot/nautobot/blob/452f02852489d60d9620ffa5fd67ab955c2181cd/nautobot/extras/models/customfields.py#L588) since that doesnt sanitize/check the `field.label` data?

# TODO

- [x] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

# SCREENSHOTS
image
image
image

### CODE REVIEW

**Comment by MeganerdDev** [comment]
Definitely open for feedback!
- I tested this on Nautobot v1.5.22b1 to be working
- I am using the SmallTextarea widget, which could instead just be the default widget

**Comment by bryanculver** [comment]
PR might look ugly until #4020 goes in.

File: nautobot/extras/models/customfields.py [hunk 1]
```diff
@@ -31,6 +31,8 @@
     StaticSelect2,
     StaticSelect2Multiple,
     add_blank_choice,
+    CommentField,
+    SmallTextarea,
 )
```
Comment (glennmatthews) [inline nautobot/extras/models/customfields.py:36]: Nitpick: let's keep the entries here in alphabetical order rather than adding to the end.

**Review by glennmatthews — COMMENTED** [review]
Looks both straightforward and useful - my favorite type of PR! :-)

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
Oh oops, looks like our CI isn't checking for it in `develop-1.6` yet - could you please add a `changes/4006.added` changelog fragment as part of this PR?

**Comment by MeganerdDev** [comment]
> Oh oops, looks like our CI isn't checking for it in `develop-1.6` yet - could you please add a `changes/4006.added` changelog fragment as part of this PR?

Thanks Glenn, you previously reminded me on another pull request on the changelog fragment. I'll try and get that jammed in my brain! Fragment `./changes/4006.added` added

**Pull request merged**